platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/CryptoStatic.java

```diff
@@ -254,22 +255,22 @@ static KeyStore createEmptyTrustStore() throws KeyStoreException {
     }
 
     /**
-     * See {@link SignatureVerifier#verifySignature(byte[], byte[], PublicKey)}
+     * See {@link SignatureVerifier#verifySignature(Bytes, Bytes, PublicKey)}
      */
     public static boolean verifySignature(
-            @NonNull byte[] data, @NonNull byte[] signature, @NonNull PublicKey publicKey) {
+            @NonNull final Bytes data, @NonNull final Bytes signature, @NonNull final PublicKey publicKey) {
         Objects.requireNonNull(data);
         Objects.requireNonNull(signature);
         Objects.requireNonNull(publicKey);
         try {
             final Signature sig = Signature.getInstance(CryptoConstants.SIG_TYPE2, CryptoConstants.SIG_PROVIDER);
             sig.initVerify(publicKey);
-            sig.update(data);
-            return sig.verify(signature);
-        } catch (NoSuchAlgorithmException | NoSuchProviderException e) {
+            data.updateSignature(sig);
+            return signature.verifySignature(sig);
+        } catch (final NoSuchAlgorithmException | NoSuchProviderException e) {
             // should never happen
             throw new CryptographyException(e);
```

Comment:
Can you add a message to that exception.

Reply:
added